### PR TITLE
Mar 1051

### PR DESCRIPTION
**Pull request by jamespacileo**

### CODE REVIEW

File: api/barriers/models.py [hunk 1]
```diff
@@ -1201,6 +1162,14 @@ def priority_filter(self, queryset, name, value):
         else:
             return queryset.filter(priority__in=priorities)
 
+    def has_action_plan_filter(self, queryset, name, value):
+        if value:
```
Comment (cad106uk) [inline api/barriers/models.py:1166]: This is a guard and should be written as 

```python
if not value:
    return queryset
```

File: api/barriers/models.py [hunk 2]
```diff
@@ -1201,6 +1162,14 @@ def priority_filter(self, queryset, name, value):
...
+            from api.action_plans.models import ActionPlanMilestone
```
Comment (cad106uk) [inline api/barriers/models.py:1167]: Why is this imported here and not at the top of the file

File: api/history/manager.py [hunk 3]
```diff
@@ -63,15 +60,16 @@ def get_public_activity(cls, public_barrier, use_cache=False):
 
     @classmethod
     def get_full_history(cls, barrier, ignore_creation_items=False, use_cache=False):
-        if use_cache:
-            return cls.get_full_history_cached(barrier, ignore_creation_items)
+        # if use_cache:
```
Comment (cad106uk) [inline api/history/manager.py:63]: This committed out code should just be deleted

File: config/settings/base.py [hunk 4]
```diff
@@ -252,6 +252,7 @@
     "PAGE_SIZE": 400,
     "DEFAULT_AUTHENTICATION_CLASSES": [
         "oauth2_provider.contrib.rest_framework.OAuth2Authentication",
+        "rest_framework.authentication.SessionAuthentication",
```
Comment (cad106uk) [inline config/settings/base.py:255]: Why is a new authentication system being added here?

**Review by cad106uk — CHANGES_REQUESTED** [review]
some feedback

File: api/barriers/models.py (hunk 2 above)
Comment (jamespacileo) [reply api/barriers/models.py:1167]: circular dependency

File: api/barriers/models.py (hunk 1 above)
Comment (jamespacileo) [reply api/barriers/models.py:1166]: done

File: api/history/manager.py (hunk 3 above)
Comment (jamespacileo) [reply api/history/manager.py]: done

File: config/settings/base.py (hunk 4 above)
Comment (jamespacileo) [reply config/settings/base.py]: removed

**Review by cad106uk — APPROVED** [review]

**Pull request merged**